internal/dag/httpproxy_processor.go

```diff
@@ -700,6 +709,27 @@ func determineSNI(routeRequestHeaders *HeadersPolicy, clusterRequestHeaders *Hea
 	return service.ExternalName
 }
 
+func corsPolicy(policy *projcontour.CorsPolicy) (*CorsPolicy, error) {
+	if policy == nil {
+		return nil, nil
+	}
+	maxAge := timeout.DefaultSetting()
+	if perTryMaxAge, err := time.ParseDuration(policy.MaxAge); err == nil {
+		maxAge = timeout.DurationSetting(perTryMaxAge)
+		if maxAge.Duration().Seconds() < 0 {
+			return nil, fmt.Errorf("invalid max age value: %q", policy.MaxAge)
+		}
+	}
```

Comment:
Do you want to support disabling this setting? Per https://developer.mozilla.org/en-US/docs/Web/HTTP/Headers/Access-Control-Max-Age, the value `-1` disables the timeout.

If so, the standard for other Contour timeouts is that the input string `infinity` disables the timeout. The `timeout.Parse` function already handles parsing this.

Reply:
> For practical purposes, a max-age of -1 and 0 seem like they would be treated similarly?

Both 0 and -1 mean disable caching for the preflight request but I decided to support just the 0 value. I found tricky to handle the -1  taking into account that the durations are expressed in the Go duration format. If we wanted to support -1 how would we do that? -1s disables caching but -1h is an invalid value? I'd rather allow negative values and assume that all of them disable the cache.